From 69c1909da075f68961276dc4ca3720ee0346ba2b Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sun, 25 Sep 2016 22:08:26 +0200
Subject: [PATCH 4/9] Prevent symbols not being exported in Qt5WebKit.dll

WebKit svn commit 136242 implemented a split into QtWebKit and QtWebKitWidgets.
Due to this change a static library named WebKit1.a is created first.
After this a shared library is created named Qt5WebKit.dll which contains
the contents of this static library and some other object files.
However, various symbols in the static library are expected to be exported
in the Qt5WebKit.dll shared library. As static libraries normally don't care
about exported symbols (everything is exported after all) the decoration
attribute Q_DECL_EXPORT won't be set.
This results in undefined references when trying to link the QtWebKitWidgets
shared library (which depends on various symbols which aren't exported properly
in the Qt5WebKit.dll shared library).
---
 Source/WebKit/qt/Api/qwebkitglobal.h | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/Source/WebKit/qt/Api/qwebkitglobal.h b/Source/WebKit/qt/Api/qwebkitglobal.h
index fe222c2fc..46d629c05 100644
--- a/Source/WebKit/qt/Api/qwebkitglobal.h
+++ b/Source/WebKit/qt/Api/qwebkitglobal.h
@@ -24,7 +24,6 @@
 #include <QtCore/qglobal.h>
 #include <QtCore/qstring.h>
 
-#ifndef QT_STATIC
 #  if !defined(QT_BUILD_WEBKITWIDGETS_LIB) && defined(BUILDING_WEBKIT)
 #      define QWEBKIT_EXPORT Q_DECL_EXPORT
 #  else
@@ -35,10 +34,6 @@
 #  else
 #      define QWEBKITWIDGETS_EXPORT Q_DECL_IMPORT
 #  endif
-#else
-#  define QWEBKITWIDGETS_EXPORT
-#  define QWEBKIT_EXPORT
-#endif
 
 QWEBKIT_EXPORT QString qWebKitVersion();
 QWEBKIT_EXPORT int qWebKitMajorVersion();
-- 
2.13.0

